types/service/http_service_configuration.go

```diff
@@ -24,6 +24,7 @@ type HttpServiceConfiguration struct {
 // configuration for standard http services (fka sockets).
 type StandardHttpServiceConfiguration struct {
 	HostnameAndPort // inherited
+	HostSniHeader   string
```

Comment:
No json tag. I think "host_header" is good enough. But maybe @atoonk has a better idea.